src/staticdata.c

```diff
@@ -1316,6 +1319,9 @@ static void jl_prune_type_cache(jl_svec_t *cache)
     }
     if (i > ins) {
         memset(&jl_svec_data(cache)[ins], 0, (i - ins) * sizeof(jl_value_t*));
+        while (l > 0 && l/2 >= ins)
+            l = l/2;
+        jl_svec_len(cache) = l == 0 ? 1 : l;
```

Comment:
```suggestion
        jl_svec_len(cache) = l == 0 ? 1 : l; // XXX: this'll cause the GC to abort later since we've corrupted the object metadata
        jl_rehash_type_cache(tn); // might as well do this now, and not on every restart?
```